Hangman/HangmanGame.py

```diff
@@ -0,0 +1,119 @@
+from Hangman import Hangman
+from WordGenerator import WordGenerator
+
+
+class HangmanGame:
+    def __init__(self) -> None:
+        self.total_points = 0
+        self.word_generator = WordGenerator()
+        self.hangman = Hangman()
+        self.word = ''
+        self.incorrect_guess_count = 0
+        self.word_display = []
+        self.guesses = []
+        self.status = ''
+
+    def start_game(self) -> None:
+        self.__set_word(self.word_generator.get_next_word())
```

Comment:
start_game is doing too many things, we should abstract it out further. 
like 17 and 18 can be moved to self.initialize_game()